src/server/engine_shard.cc

```diff
@@ -716,7 +742,12 @@ void EngineShard::RetireExpiredAndEvict() {
     ttl_delete_target = kTtlDeleteLimit * double(deleted) / (double(traversed) + 10);
   }
 
-  ssize_t eviction_redline = size_t(max_memory_limit * kRedLimitFactor) / shard_set->size();
+  // For memory we are using threshold for free memory
+  const ssize_t shard_memory_budget_threshold =
+      ssize_t(max_memory_limit * kEvictionFreeMemoryThreshold) / shard_set->size();
+
+  // Calculate how much rss memory we need to free per shard
+  ssize_t rss_memory_to_free = CalculateRssMemoryToFreePerShard();
```

Comment:
in the for loop below we call DeleteExpiredStep which can also remove entries, so calculating the rss memory to free here and updating it with released_bytes is not enough.

Reply:
Yes, I understand it